config/setup.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	pb "go.viam.com/api/service/slam/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// SetupDirectories creates the data directory at the specified path along with
+// its data, map, and config subdirectories.
+func SetupDirectories(dataDirectory string, logger golog.Logger) error {
+	for _, directoryName := range [4]string{"", "data", "map", "config"} {
+		directoryPath := filepath.Join(dataDirectory, directoryName)
+		if _, err := os.Stat(directoryPath); err != nil {
```

Comment:
Why was this changed to be any error?

Reply:
This was changed to be any error to prevent a silent failure if the function was unable to stat the directory for other reasons (ex: the directory is not readable)